src/workers/newNotificationV2Mail.ts

```diff
@@ -151,21 +151,29 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
   source_post_submitted: async (con, user, notification) => {
     const moderation = await con.getRepository(SourcePostModeration).findOne({
       where: { id: notification.referenceId },
-      select: ['sourceId', 'image', 'title', 'content', 'type', 'sharedPostId'],
+      select: [
```

Comment:
Can you type the `moderation` with `Pick` so it properly shows what fields are available.

Reply:
That was part of the issue because typeorm select is not applied over findOne type.

Reply:
Also on below `select` queries in Promise.all

Reply:
Makes sense. This is very interesting. I might actually check if we can contribute to this part of the type in the package 👀

Reply:
that would be nice, they do offer it as generic with QueryBuilder and `getRawMany` but you still have to `Pick` by hand